Tensile/KernelWriterAssembly.py

```diff
@@ -7611,56 +7815,92 @@ def localReadDo(self, kernel, bufferIdx, iui, epsi, uIdx, tP):
     offsetMultiplier = 1 # instruction.offsetMultiplier
     #totalReads = (kernel["ThreadTile%u"%tP["tensorIdx"]]/blockWidth) / numOffsets
     valuIdx = 0
+
+    # cleanup code 
+    # simplify both f32, half/float16 code ; no need for branch (next optimization)
+    # remove unnecessary numReadsAlongK (numReadPerVector should take care of that)
+    # numVectorsPerTile = take care of numer of instructions per wave
+    # numReadsPerVector = take care of number of local reads required per instruction
+    # re-insert numberofOffset loop ; next optimization to fetch different colIdx/rowIdx for broadcast/swizzling option
+    # next clean-up 
```

Comment:
Agree with removing the extra var `numReadsAlongK` and half-type specific branch, as the original concept `numVectorsPerTile/numReadsPerVector/numOffsets` cover this already